mesonbuild/interpreter.py

```diff
@@ -2263,6 +2264,42 @@ def _get_native() -> str:
             else:
                 return _get_native()
 
+    @permittedKwargs({'encoding', 'allow_absolute'})
+    @FeatureNew('meson.get_file_contents', '0.58.0')
+    def get_file_contents_method(
+            self, args: T.Sequence[str], kwargs: dict) -> str:
+        '''
+        Read a file from the source tree and return its value as a decoded
+        string. If the encoding is not specified, the file is assumed to be
+        utf-8 encoded. Paths must be relative by default (to prevent accidents)
+        and are forbidden to be read from the build directory (to prevent build
+        loops)
+        '''
+        if len(args) != 1:
+            raise InterpreterException('expected single positional <path> arg')
```

Comment:
A `mesonlib.File` object should be acceptable as well.

Reply:
done. Please check the logic here (especially around `subdir` handling. I added this to the tests, but I'd like it double-checked